Purchases/Public/RCPurchases.m

```diff
@@ -334,10 +342,21 @@ - (void)checkTrialOrIntroductoryPriceEligibility:(NSArray<NSString *> *)productI
     }];
 }
 
-- (void)makePurchase:(SKProduct *)product
+- (void)makePurchase:(SKProduct *)product withCompletionBlock:(RCPurchaseCompletedBlock)completion
 {
     SKMutablePayment *payment = [SKMutablePayment paymentWithProduct:product];
     payment.applicationUsername = self.appUserID;
+    
+    @synchronized (self) {
+        if (self.purchaseCompleteCallbacks[product.productIdentifier]) {
+            completion(nil, nil, [NSError errorWithDomain:RCPurchasesAPIErrorDomain
+                                                     code:RCDuplicateMakePurchaseCallsError
+                                                 userInfo:@{
+                                                            NSLocalizedDescriptionKey: @"Purchase already in progress for this product."
+                                                            }]);
+        }
+        self.purchaseCompleteCallbacks[product.productIdentifier] = [completion copy];
```

Comment:
I think this should be inside an `else`. If the if is satisfied, the completion block passed to this method will be triggered with an error and then the old completion in the dict will be overriden with the completion block passed to this method (which has already been called with an error).

Reply:
Good catch.